File: s3/1번/quiz/solution/python/solution.py
import boto3

# LocalStack 설정
LOCALSTACK_ENDPOINT = "http://localhost:4566"
AWS_REGION = "us-east-1"
bucket_name = "my-bucket"
file_key = "problem.txt"

# boto3 S3 클라이언트 생성
s3 = boto3.client(
    "s3",
    endpoint_url=LOCALSTACK_ENDPOINT,
    region_name=AWS_REGION,
    aws_access_key_id="test",
    aws_secret_access_key="test"
)

# S3 객체 가져오기
response = s3.get_object(Bucket=bucket_name, Key=file_key)

# Body (스트림) → 문자열로 변환
content = response["Body"].read().decode("utf-8")

print(f"📄 Content of {file_key}:\n{content}")

File: s3/1번/quiz/solution/javascript/solution.js
import { S3Client, GetObjectCommand } from "@aws-sdk/client-s3";

// 스트림을 문자열로 변환하는 유틸 함수
async function readableStreamToString(stream) {
  const chunks = [];
  for await (const chunk of stream) {
    chunks.push(Buffer.from(chunk));
  }
  return Buffer.concat(chunks).toString("utf-8");
}

// LocalStack S3 클라이언트 설정
const s3 = new S3Client({
  region: "us-east-1",
  endpoint: "http://localhost:4566", // ngrok 쓰는 경우는 여기 URL 변경
  credentials: {
    accessKeyId: "test",
    secretAccessKey: "test",
  },
  forcePathStyle: true,
});

const bucketName = "my-bucket";
const fileKey = "problem.txt";

async function readProblemFile() {
  const response = await s3.send(
    new GetObjectCommand({
      Bucket: bucketName,
      Key: fileKey,
    })
  );

  const content = await readableStreamToString(response.Body);
  console.log(`📄 Content of ${fileKey}:\n${content}`);
}

await readProblemFile();
